src/OpenLoco/Windows/TextInputWindow.cpp

```diff
@@ -41,6 +42,20 @@ namespace OpenLoco::Ui::Windows::TextInput
         };
     }
 
+    std::map<string_id, int16_t> inputLengthLimit{
+        { StringIds::title_name_owner, 31 },
+        { StringIds::title_name_company, 31 },
+        { StringIds::preferred_owner_name, 31 },
+        { StringIds::scenario_name_title, 63 },
+        { StringIds::scenario_details_title, 199 },
+        { StringIds::title_station_name, 31 },
+        { StringIds::chat_title, 31 },
+        { StringIds::title_industry_name, 31 },
+        { StringIds::enter_host_address, 15 },
+        { StringIds::title_town_name, 31 },
+        { StringIds::title_name_vehicle, 31 }
+    };
+
```

Comment:
This feels a little hacky. I think it would be better to have the invoking window pass the limit, rather than mapping limits based on a string id.

Reply:
Agreed

Reply:
And all the 31s should use the kMaxUserStringLength-1 or whatever we named the constant

Reply:
> And all the 31s should use the kMaxUserStringLength-1 or whatever we named the constant

Perhaps we can even make this the default, as it seems to be the most common.